### ISSUE TITLE
Add Versioning Scheme to API

### ISSUE THREAD

**Issue by te0d** [ipfs/ipfs-cluster#193]
Currently the API is exposed by default at "/ip4/127.0.0.1/tcp/9094" so that, for example, `curl http://127.0.0.1:9094/id` will return information about your node. `go-ipfs` appears to be using a versioning scheme so that the API is prefixed with "../api/v0/id". I belive this could become useful for future modifications to the API to prevent breaking changes (once out of alpha).

**Comment by hsanjuan** [comment]
Hi,

yes, this is something that I've thought of myself. [this post](https://www.troyhunt.com/your-api-versioning-is-wrong-which-is/) illustrates some of the problems with different approaches to API versioning, with pros and cons.

There is a fight between:
  * The fact that APIs will change in breaking ways at some point, and gotta do something about it
  * The fact that maintaining different versions of the API alive, in parallel is a terrible overhead (see how even Facebook deprecates old api versions and shuts the endpoints down).
  * The necessity of keeping APIs stable as long as possible

`go-ipfs` includes a `v0` but it's not aiming to be a REST API nor nothing indicates there will ever be a v1, or that v0 will be maintained alongside an eventual `v1`. It could also not have `/v0/` at all, and if ever it changes, introduce a `/v1/` for the new version.

In the case of cluster, the `rest` API cluster component provides these endpoints. In case we wanted to change them, we could, for example, have a new api component and use it instead (or use both on different ports, or make a router for both). Or we could choose a versioning strategy and implement it on the current component. Or we could just say that cluster APIs are gonna change and do a major version bump.

In short:

* Versioning strategies for the API can be introduced anytime, if the need arises (by any of the methods in the article, leaving the current endpoints as they are for compatibility if we decide so).
*  We will possibly be faced with this question, but the decision will depend on variables unknown at that point: type cluster usage and deployments, existing applications on top of cluster, type of API consumers and number of developers involved in the project.

So, ***right now***, we don't have to make a decision on how the API will be versioned. I think it is good to leave this issue open for further discussion on this. I'm not completely sold on the `/v1/` way, but I agree it is explicit and user-friendly (which are good things for an API, even if it's not RESTrict).

**Comment by te0d** [comment]
Yes, I agree that it's something that is not necessary right now. And debatable as to how/if it should be done based on the direction of the project. Thanks for the thought-out response!

**Comment by hsanjuan** [comment]
Related #452
